feat(label): refactored label component (#2943)
patternfly/patternfly: src/patternfly/components/Label/examples/Label.md

import './Label.css'

## Examples
```hbs title=Filled
{{#> label label--id="default-grey"}}
  {{#> label-text}}
    grey label
  {{/label-text}}
{{/label}}

> {{#> label label--id="default-grey-icon"}}

Reviewer: did you include a compact variation in here?

  {{#> label-icon}}
    <i class="fas fa-info-circle" aria-hidden="true"></i>
  {{/label-icon}}
  {{#> label-text}}
    cyan label
  {{/label-text}}
{{/label}}

{{#> label label--id="outline-cyan" label--modifier="pf-m-cyan pf-m-outline pf-m-hoverable" label--attribute='tabindex="0"'}}
  {{#> label-text}}
    cyan hoverable label
  {{/label-text}}
{{/label}}
```

## Documentation
### Overview
Labels can be used in a variety of components and can adjust in font size to match that of the component it lives in. For example, labels can be used in tables. Specifically, the compact table has a modifier that adjusts its font size, so when using a label in this table, it's important to also add its respective `.pf-c-compact` modifier.

### Usage
| Class | Applied to | Outcome |
| -- | -- | -- |
| `.pf-c-label` | `<span>` | Iniates a label. Without a color modifier, the label's default style is grey. Use a color modifier to change the label color.  |
| `.pf-m-compact` | `.pf-c-label` | Modifies label for compact styles. |
| `.pf-m-outline` | `.pf-c-label` | Modifies label for outline styles. |
| `.pf-m-hoverable` | `.pf-c-label` | Modifies label for a hoverable styles. |

> | `.pf-m-blue` | `.pf-c-label` | Modifies label for a blue styles. |

Reviewer: ```suggestion
| `.pf-m-blue` | `.pf-c-label` | Modifies the label to have blue colored styling. |
```
